### ISSUE TITLE
Diagnostics suggested by new testing framework apparently insert wrong spans

### ISSUE THREAD

**Issue by taori** [dotnet/roslyn-sdk#420]
After applying the changes you suggested in #418 i had to add the other warnings as well - It told me i had missing diagnostics so i added the ones listed in output. However that tells me i have the wrong spans.

![grafik](https://user-images.githubusercontent.com/5545184/72454554-5f946d00-37c1-11ea-8503-edbb02999c1a.png)

**Comment by sharwell** [comment]
CS1591 is added to the `TestState` expected diagnostics, and then the default behavior is to copy it to the `FixedState` expected diagnostics because there is no registered code fix for this diagnostic. If you explicitly set the expected diagnostics of `FixedState`, it will use that instead of inheriting the expectations from `TestState`.

**Comment by taori** [comment]
I see. That really increases the necessity for #419. If i add the generated statements and omit the diagnostic in the FixedState.ExpectedDiagnostics property i will never get any useful information from the test output, on how to fix my CodeFix, but rather am being yelled at, that the number of diagnostics is inacurate (which is correct obviously) as my CodeFix is not applied yet in a proper way.

I suppose since #419 is tagged as help wanted i can just provide a PR and this will be a temporary issue once you can restrict the listing of diagnostics to the ones you are actually interested in fixing?

Thanks for getting me started on this scenario though!

**Comment by taori** [comment]
If #421 is merged this can be closed.

**Comment by sharwell** [comment]
I'm going to mark this as **wontfix**. While it's theoretically possible to automatically adapt the expected results to new locations based on changes made by the fix, these algorithms sometimes make mistakes and tend to execute slowly. The recommended approach for dealing with this situation is explicitly specifying the expected diagnostics for the fixed code.

**Comment by sharwell** [comment]
@taori While this issue won't be directly fixed, the problem will go away on its own once the improved filtering is in place. 👍
